src/io.cr

```diff
@@ -350,12 +350,30 @@ module IO
   # Writes the formatted string to this IO. The format can include
   # format specifiers(subsequences beginning with %). The arguments
   # replace the format specifiers in the format string.
+  # The specifiers that can be supplied to the method are
+  # %s => strings
+  # %b => binary
+  # %o => octal
+  # %d => decimal
+  # %i => decimal
+  # %x => hexadecimal
+  # %X => hexadecimal
+  # %a, %A, %e, %E, %f, %g, %G => char
```

Comment:
@jhass What is the representation of the value called after it is formatted by `%a` or `%A`. I got these results while trying out.

```crystal
io = MemoryIO.new
io.printf("Hello %a", "4")
p io.to_s # => "Hello 0x1p+2"
```
What is this form `0x1p+2` called?